go/tools/builders/nogo_main.go

```diff
@@ -172,6 +172,21 @@ func checkPackage(analyzers []*analysis.Analyzer, packagePath string, packageFil
 
 	roots := make([]*action, 0, len(analyzers))
 	for _, a := range analyzers {
+		if cfg, ok := configs[a.Name]; ok {
+			for flagKey, flagVal := range cfg.analyzerFlags {
+				if err := a.Flags.Set(flagKey, flagVal); err != nil {
+					if strings.HasPrefix(flagKey, "-") {
+						return "", nil, fmt.Errorf(
+							"%s: flag should not begin with '-': %s", a.Name, flagKey)
+					}
+					if flag := a.Flags.Lookup(flagKey); flag == nil {
+						return "", nil, fmt.Errorf("%s: unrecognized flag: %s", a.Name, flagKey)
+					}
+					return "", nil, fmt.Errorf(
+						"%s: invalid value for flag: %s=%s", a.Name, flagKey, flagVal)
+				}
```

Comment:
The message tends to be pretty generic and unhelpful. Maybe it would make sense to include the error if it had something other than that generic message.

Reply:
On second thought, the message helps distinguish between range errors and other errors at least, so it might be worth keeping.

Reply:
I don't mind taking another pull request if you want to log the error.

Reply:
Ah sorry, I already added it. I just forgot to declare my change of heart here. I'll mark this thread as resolved.